src/LogObject.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace Avalara;
+
+class LogInformation
+{
+    private $startTime;
+    private $includeReqResInLogging;
+    public $httpMethod;
+    public $headerCorrelationId;
+    public $requestDetails;
+    public $responseDetails;
+    public $requestURI;
+    public $totalExecutionTime;
+    public $statusCode;
+    public $timestamp;
+    public $exceptionMessage;
+
+    public function setStartTime($startTime)
+    {
+        $this-> startTime = $startTime;
+    }
+
+    public function populateRequestInfo($verb, $apiUrl, $guzzleParams, $includeReqResInLogging)
+    {
+        $this-> timestamp = gmdate("Y/m/d H:i:s");
+        $this-> httpMethod = $verb;
+        $this-> requestURI = $apiUrl;
+        $this-> includeReqResInLogging = $includeReqResInLogging;
+        if($this-> includeReqResInLogging)
+        {
+            $this-> requestDetails = $guzzleParams['body'];
+        }
+    }
+
+    public function populateResponseInfo($jsonBody, $response)
+    {
+        $this-> populateCommonResponseInfo($response);
+        if($this-> includeReqResInLogging)
+        {
+            $this-> responseDetails = $jsonBody;
+        }
+    }
+
+    public function populateErrorInfoWithMessageAndBody($errorMessage, $response) 
+    {
+        $this-> populateCommonResponseInfo($response);
+        $this-> exceptionMessage = $errorMessage;
+        $this-> statusCode = 500;
```

Comment:
Do we want to put statusCode 500 when its actually a json parse error in some cases? maybe it should be zero or null?

Reply:
Yes and my take on this was that it was throwing a general Exception earlier and I wanted to keep the same behavior here, and for that 500 seemed valid in this case. This is how it was earlier, FYR:
![Screen Shot 2022-11-02 at 11 35 57 AM](https://user-images.githubusercontent.com/59040125/199574257-07294ef8-a655-4f74-a486-231fb735c741.png)

Reply:
I think maybe we should put the http status code what the actual response code is, and then add the error into the error info. Reason being it would be confusing if a customer is contacting avatax support with a correlationid saying, we see a response code 500 in the logs, but Avatax support will most likely have some different http code in their logs because we are not using the actual code

Reply:
As per our discussion, we will keep it this way for now and if we get proposals at later point in time then we will circle back and improve it based on proposals/feedback.